### PR TITLE
feat(queue): add support for update_method and update_bot_account

### PR DESCRIPTION
**Pull request by sileht**
Fixes MRGFY-507

Change-Id: I33c7ce783b0f8a58fd847f178f050dccb0a23acc

### CODE REVIEW

File: docs/source/actions/queue.rst [hunk 1]
```diff
@@ -224,6 +224,29 @@ These are the options of the ``queue`` action:
...
+     -
+       Method to use to update the pull request with its base branch when the
+       speculative checks is done ``inplace``.
+       Possible values:
+
+       * ``merge`` to merge the base branch into the pull request.
+       * ``rebase`` to rebase the pull request against its base branch.
+
+       Note that the ``rebase`` method has some drawbacks, see :ref:`update method rebase`.
+
```
Comment (jd) [inline docs/source/actions/queue.rst:239]:
```suggestion
     - Method to use to update the pull request with its base branch when the
       speculative check is done in-place.
       Possible values:

       * ``merge`` to merge the base branch into the pull request.
       * ``rebase`` to rebase the pull request against its base branch.

       Note that the ``rebase`` method has some drawbacks, see :ref:`update method rebase`.
```

File: docs/source/actions/queue.rst [hunk 2]
```diff
@@ -421,4 +444,34 @@ request:
 
     This feature does not work for cross-repository dependencies.
 
+.. _update method rebase:
+
+Rebase as update_method
```
Comment (jd) [inline docs/source/actions/queue.rst:449]:
```suggestion
Using Rebase to Update
```

File: docs/source/actions/queue.rst [hunk 3]
```diff
@@ -421,4 +444,34 @@ request:
...
+* GPG signed commits will lost their signatures.
+
+* Mergify will use a token from one of the repository member to force-push the
+  branch. GitHub Applications are not allowed to do that so. This is a GitHub
+  limitation that we already have reported — there is not much Mergify can do
+  about that. In order to make this works, Mergify randomly picks and borrows a
+  token from one of your repository member and uses it to force-push the
+  rebased branch. If you need to control that looks at ``update_bot_account`` option.
+  The GitHub UI will show your collaborator as the author of
+  the push, while it actually has been executed by Mergify.
```
Comment (jd) [inline docs/source/actions/queue.rst:475]:
```suggestion
Using the ``rebase`` method to update a pull request with its base branch has some
drawbacks:

* It doesn't work for private forked repositories.

* Every commits SHA-1 of the pull request will change. The pull request author
  will need to force-push its own branch if they add new commits.

* GitHub branch protections of your repository may dismiss approved reviews.

* GitHub branch protection of the contributor repository may deny Mergify
  force-pushing the rebased pull request.

* GPG signed commits will lose their signatures.

* Mergify will impersonate one of the repository members to force-push the
  branch as GitHub Applications are not authorized to do that by themselves.
  If you need to control which user is impersonated, you can use the ``update_bot_account`` option.
  Be aware that the GitHub UI will show the collaborator as the author of the push, while it was actually executed by Mergify.
```

File: releasenotes/notes/queue-update-method-17d3be1f0ccc8b03.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    The ``queue`` action got two new options ``update_method`` and ``update_bot_account``.
```
Comment (jd) [inline releasenotes/notes/queue-update-method-17d3be1f0ccc8b03.yaml:4]:
```suggestion
    The ``queue`` action got two new options: ``update_method`` and ``update_bot_account``.
```

**Review by jd — CHANGES_REQUESTED** [review]

**Comment by sileht** [comment]
@mergifyio squash first_commit

**Comment by sileht** [comment]
@Mergifyio squash first-commit

**Review by jd — APPROVED** [review]

**Pull request merged**